### PR TITLE
update docker image builds to publish `3-latest` images

### PR DESCRIPTION
**Pull request by zzstoatzz**
closes https://github.com/PrefectHQ/prefect/issues/14228

allows ad-hoc publishing of `3-latest` images via `workflow_dispatch`

this PR:
- adds the ability to publish `3-latest` tags (for each minor version and flavor) on `workflow_dispatch`

### CODE REVIEW

File: .github/workflows/docker-images.yaml [hunk 1]
```diff
@@ -23,8 +23,14 @@ on:
       - ".github/workflows/docker-images.yaml"
       - "ui/**"
 
-  # On workflow_dispatch, push sha and branch patterns to prefect-dev
+  # On workflow_dispatch, allow publishing 3-latest images
   workflow_dispatch:
+    inputs:
+      publish_3_latest:
+        description: 'Publish 3-latest images'
+        required: true
```
Comment (cicdw) [inline .github/workflows/docker-images.yaml:31]: why is this required if it defaults to `false`?

Comment (zzstoatzz) [reply .github/workflows/docker-images.yaml]: good point! updated in [bb80382](https://github.com/PrefectHQ/prefect/pull/14633/commits/bb80382718262e3f7d8977cbb5fe111d2ebaf376)
